google/cloud/bigquery/external_config.py

```diff
@@ -693,6 +694,28 @@ def compression(self):
     def compression(self, value):
         self._properties["compression"] = value
 
+    @property
+    def decimal_target_types(self) -> Optional[FrozenSet[str]]:
+        """Possible SQL data types to which the source decimal values are converted.
+
+        See:
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#ExternalDataConfiguration.FIELDS.decimal_target_types
+
+        .. versionadded:: 2.21.0
+        """
+        prop = self._properties.get("decimalTargetTypes")
+        if prop is not None:
+            prop = frozenset(prop)
```

Comment:
The enum docstrings added above suggest that order is significant.  However, `frozenset` is unordered:  if the ordering is signifcant, then we should be preserving it -- maybe return a tuple to preserve the order?

Reply:
As per [docs](https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#ExternalDataConfiguration.FIELDS.decimal_target_types), the order is not significant, which why a (frozen) set is returned.

> The order of the types in this field is ignored.

I'll re-read the enum docstrings and make this more clear, if it's currently ambiguous.

Reply:
So we are storing it as a list in the setter for compatibility with how the back-end sends it to us?

Reply:
It's primarily because sets are not JSON serializable out of the box - `json.dumps()` uses `JSONEncoder` by default, but the latter doesn't know how to serialize sets.

Reply:
@shollyman The `DecimalTargetType` enum [docs](https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#DecimalTargetType) can be misleading if not knowing the entire context. That _"Depending on the order"_ bit might be understood as if the order matters when providing a list of target decimal types, thus I removed it from the Python class docstrings.

Should the generated docs follow suit? Or perhaps modify the wording?

Reply:
Yeah, order absolutely matters here.  The feedback about the docs is good, but I'm going to punt that to @stephaniewang526 to drive.

Reply:
Hi Peter, yes I can see how the doc is a little confusing... Order matters very much in this feature please.

Reply:
"disco doc" - gotta love that lingo. :heart: :confetti_ball: :rofl:

Reply:
Upon checking w myzhong@, the order actually does not matter here. It would always go for NUMERIC if it's possible even if user specified [\"BIGNUMERIC\", \"NUMERIC\"].

Reply:
OK thanks, we'll keep the target types in a `frozenset` then. We should probably reword the generated enum docs, though.

Reply:
b/193235792 created to track the doc update. cl/383854927 created to update the doc. Pending review from BQ techwriters.